gapis/resolve/pipeline.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolve
+
+import (
+	"context"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/messages"
+	"github.com/google/gapid/gapis/service"
+	"github.com/google/gapid/gapis/service/path"
+)
+
+// Pipelines resolves the data of the currently bound pipelines at the specified
+// point in the capture.
+func Pipelines(ctx context.Context, p *path.Pipelines, r *path.ResolveConfig) (interface{}, error) {
+	obj, err := ResolveInternal(ctx, p.Parent(), r)
+	if err != nil {
+		return nil, err
+	}
+
+	if pipelines, err := pipelinesFor(ctx, obj, p, r); err != api.ErrPipelineNotAvailable {
+		return pipelines, err
+	}
+	return nil, &service.ErrDataUnavailable{Reason: messages.ErrNotADrawCall()}
+}
```

Comment:
Here and in other places, the api.ErrPipelineNotAvailable is replaced by messages.ErrNotADrawCall, I'm not sure why we don't simply forward the api.ErrPipelineNotAvailable error?

Reply:
`ErrPipelineNotAvailable` is not a "service error" and would be wrapped inside an `ErrInternalError`. However, we want to return the `ErrDataUnavailable` error here, because it has a specific semantic meaning to the client. `ErrPipelineNotAvailable` is not so much an "error" but a "null" indicator, kind of like `io.EOF`, i.e. not unexpected.